streams/src/test/java/org/apache/kafka/streams/state/internals/MeteredWindowStoreTest.java

```diff
@@ -18,99 +18,69 @@
 import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertTrue;
 
 public class MeteredWindowStoreTest {
     private InternalMockProcessorContext context;
     @SuppressWarnings("unchecked")
     private final WindowStore<Bytes, byte[]> innerStoreMock = EasyMock.createNiceMock(WindowStore.class);
-    private final MeteredWindowStore<String, String> store = new MeteredWindowStore<>(innerStoreMock, "scope", new MockTime(), Serdes.String(), new SerdeThatDoesntHandleNull());
-    private final Set<String> latencyRecorded = new HashSet<>();
-    private final Set<String> throughputRecorded = new HashSet<>();
+    private final MeteredWindowStore<String, String> store = new MeteredWindowStore<>(
+        innerStoreMock,
+        "scope",
+        new MockTime(),
+        Serdes.String(),
+        new SerdeThatDoesntHandleNull()
+    );
+
+    {
+        EasyMock.expect(innerStoreMock.name()).andReturn("mocked-store").anyTimes();
+    }
 
     @Before
-    public void setUp() throws Exception {
-        final Metrics metrics = new Metrics();
-        final StreamsMetrics streamsMetrics = new StreamsMetrics() {
-
-            @Override
-            public Map<MetricName, ? extends Metric> metrics() {
-                return Collections.unmodifiableMap(metrics.metrics());
-            }
-
-            @Override
-            public Sensor addLatencyAndThroughputSensor(String scopeName, String entityName, String operationName, Sensor.RecordingLevel recordLevel, String... tags) {
-                return metrics.sensor(operationName);
-            }
-
-            @Override
-            public void recordLatency(final Sensor sensor, final long startNs, final long endNs) {
-                latencyRecorded.add(sensor.name());
-            }
-
-            @Override
-            public Sensor addThroughputSensor(String scopeName, String entityName, String operationName, Sensor.RecordingLevel recordLevel, String... tags) {
-                return metrics.sensor(operationName);
-            }
-
-            @Override
-            public void recordThroughput(Sensor sensor, long value) {
-                throughputRecorded.add(sensor.name());
-            }
-
-            @Override
-            public void removeSensor(Sensor sensor) {
-                metrics.removeSensor(sensor.name());
-            }
-
-            @Override
-            public Sensor addSensor(String name, Sensor.RecordingLevel recordLevel) {
-                return metrics.sensor(name);
-            }
-
-            @Override
-            public Sensor addSensor(String name, Sensor.RecordingLevel recordLevel, Sensor... parents) {
-                return metrics.sensor(name);
-            }
-
-        };
+    public void setUp() {
+        final Metrics metrics = new Metrics(new MetricConfig().recordLevel(Sensor.RecordingLevel.DEBUG));
+        final StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(metrics, "test");
```

Comment:
I replaced the override-with-capture strategy in this test with just a regular StreamsMetrics and verifying the invocation by checking that the total metric is == 1.